### DISCUSSION TITLE
Questions about Fastify

### DISCUSSION THREAD

**Discussion by thefrana** [fastify/fastify#3172] [General] [answered]
Hi,
I have a set of some questions that raised in my head. I created only one GH Discussion, because I did not want to spam with a lot of posts. First, I wanted to say that Fastify is awesome. I really like the idea of plugins. I tried NestJS before, but strong OOP approach is not for me. I also don't like boilerplate code. Fasitfy is excellent in this regard. I created a plugin that bootstraps everything and I only need to run that one file which imports the plugin from my library. The speed is amazing, of course, and also the simple "DI" with decorators is really good and easily understandable. 

1. Why is Fastify relatively small?

I looked at several sources to compare the size of the project (and I know it still does not mean anything). [NPM Trends](https://www.npmtrends.com/fastify-vs-hapi-vs-koa-vs-restify) show it's bigger than Hapi or Restify. Express and Koa are bigger, but that's understandable. So people are using Fastify. It also has nearly 20k stars on GitHub, but only a few hundreds on its plugins. On [StackOverflow](https://stackoverflow.com/questions/tagged/fastify), the tag `[fastify]` has 261 questions, which I would say is pretty small number. And according to GH, it is used by 15k people, while Restify has 32k and Hapi 20k. Again, I know it's just numbers, but I would expect these numbers to be at least twice or more bigger than they currently are.

2. Why is it not written in TypeScript?

I would say that right now, the trend in JS world is to rewrite everything in TS, but Fastify seems not to go in this way. Why? I write Fastify only with TS. The TS support is great but it can be seen that it was not primarily written for that purpose. Compared to Adonis or NestJS for example, they have TS is their core and I think it's even harder to write plain JS in these frameworks than  to use TS.

3. Is Fastify future-proof?

I pushed Fastify to be our primary backend framework at the startup where I work. I know that this is pretty hard question, but is there anything anyone can say to calm down my inner fears that it will go down like it happened with a lot of OSS projects (especially in JS ecosystem)? I don't know if you guys have solid funding or how can I (or anyone else) support you?

Thanks a lot for answers.

**Comment by RafaelGSS** [comment, marked as answer]
> On StackOverflow, the tag [fastify] has 261 questions, which I would say is pretty small number. And according to GH, it is used by 15k people, while Restify has 32k and Hapi 20k. Again, I know it's just numbers, but I would expect these numbers to be at least twice or more bigger than they currently are.

I would not use this metric to measure how big is a project, for instance, we don't have anybody from the team looking directly at StackOverflow questions as far I know. Most of the questions were done in our [discord server](https://discord.gg/D3FZYPy) and in our issues. Also, there is no problem with being small and I don't think that we need to measure the community adoption by this. As you said, it does not mean anything and likely there is no answer for your question: "Why is Fastify relatively small?".

> Why is it not written in TypeScript?

I will leave this question for other members, however, there is no problem in write anything in pure JS, and is completely your preference to write your code in TS, there are a lot of companies and developers which choose to write a pure JS implementation, I don't think that this is a problem in Fastify since we are exporting all types.

**Reply by Eomm** [reply]
I'm the StackOverflow monitor here https://stackoverflow.com/tags/fastify/topusers 😄  but I skip all the nestjs questions..

**Comment by mcollina** [comment]
> 
> 
> 1. Why is Fastify relatively small?

Fastify has been around for less time. We have been focusing on growing a community of users/maintainers instead of massive amount of end users. We are aiming for the stability of the framework and not to "get big fast" - I would say that becoming "big" is a non-goal.

An example: instead of writing features, plugins, new major releases and docs, we could have been focusing on producing tutorials and other form of promotion.

Luckily, this is starting to happen naturally now.

I would ask: how many people are actively working on Express, Restify or Koa? Are they innovating and shipping new major versions? 

> 2. Why is it not written in TypeScript?

I do not think it's possible to write Fastify in TypeScript as it heavily relies on prototypical inheritance.

> 3. Is Fastify future-proof?

The key pillar of this community is: "if you have a bug, please send a pull request". This is the path we have chosen to be future-proof, let the users make it future-proof.

> I pushed Fastify to be our primary backend framework at the startup where I work. I know that this is pretty hard question, but is there anything anyone can say to calm down my inner fears that it will go down like it happened with a lot of OSS projects (especially in JS ecosystem)?

OSS projects go down when people stop contributing. Start contributing back and help out. There are infinite amount of activities to do, from triaging to bugfixing, to implement new semver-major features. 

> I don't know if you guys have solid funding or how can I (or anyone else) support you?

We do not have any need for funding at the moment.

**Comment by thefrana** [comment]
Thanks everybody! I will mark the first post as an answer, but both of them were inspiring.
